### PR TITLE
Self Surgery + Surgery Delay Removal

### PR DESCRIPTION
**Pull request by Iamgoofball**
Self Surgery returns

All surgeries had their delays removed

Have fun enjoying surgery again!
x6n8vva

### CODE REVIEW

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 1]
```diff
@@ -18,9 +18,9 @@
 		take_organ_damage(10)
 
 /mob/living/carbon/attackby(obj/item/I, mob/user, params)
-	if(lying)
+	if(lying || user == src)
```
Comment (RemieRichards) [inline code/modules/mob/living/carbon/carbon_defense.dm]: >surgery on yourself just standing up.

What.

Comment (Iamgoofball) [inline code/modules/mob/living/carbon/carbon_defense.dm]: only way to do it nerd

Comment (RemieRichards) [inline code/modules/mob/living/carbon/carbon_defense.dm]: no it's stupid.

**Comment by xxalpha** [comment]
Not sure about removing the delays completely.

**Comment by RemieRichards** [comment]
Yeah the delay removal is completely stupid.

**Comment by Zilenan91** [comment]
I think it's kinda funny that ripping peoples hearts out with surgery is a viable tactic now.

**Comment by GunHog** [comment]
This way seems so....broken...what about a programmable AutoDoc machine or something (emag it for FUN)? It just does not make any sense just doing it on your without SOMETHING helping you...

**Comment by Iamgoofball** [comment]
You realize there's already a delay, right? There's the delay of inventory management to pull out the 10 bazillion tools. No need for a extra delay. It's just shit

**Comment by RemieRichards** [comment]
No, halve the times, but leave them in.

**Comment by Iamgoofball** [comment]
No. You missed what I said about there already being a delay due to inventory management. The do_after delays were just a pain in the ass.

Have you tried doing surgery recently? It's a pain of waiting on goddamn progress bars and then a fucking meme inventory management. I removed 1 of the pains.

**Comment by RemieRichards** [comment]
It's not a pain at all, surgery is supposed to take time, tbh the current timings are FINE (which is why they've lasted blood aeons) but since you're NOT GOING TO DROP THIS BECAUSE THATS JUST WHO YOU ARE, I said halve them.

**Comment by Iamgoofball** [comment]
It taking time isn't fun.

**Comment by Bawhoppen** [comment]
Timing makes sense WHEN ITS FUCKING SURGERY

**Comment by Anonus** [comment]
Don't just remove the delay, come on. Reduce it, maybe.

**Comment by Iamgoofball** [comment]
Please explain how beneficial surgery is that you can argue FROM A GAMEPLAY PERSPECTIVE that it deserves such a heavy downside.

**Comment by RemieRichards** [comment]
>a small wait
>heavy downside

I'm sorry Goof are you smoking all the chems you've added to the game?

**Comment by Bawhoppen** [comment]
Self surgery is dumb, and surgery should take time.

**Comment by Iamgoofball** [comment]
1. yes
2. I'm serious, it's literally TEDIOUS to do surgery for the limited benefit

@bawhoppen then how come every player ever threw a temper tantrum when self surgery was removed?

And can you argue how overpowered surgery is that it needs the balancing act of time delays?

**Comment by Bawhoppen** [comment]
thats because the people who got mad were powergaming, and they were the vocal ones

**Comment by RemieRichards** [comment]
Surgery fixes negatives, allows you to change your face, -ruins xenomorphs already-, slows down the roboticist going on their mecha rampage which YES, is a good thing.

**Comment by Zilenan91** [comment]
Yeeeaaah, perhaps make it so xeno embryo removal isn't a thing, because any cuck with a toolbelt will saw their chest open in the field to get it out and then jump right back into the fight.

**Comment by Iamgoofball** [comment]
I can blacklist implant removal and embryo removal from the self-surgery method.

Keep in mind though @Zilenan91 that by default, if you're surgerizing on the field, and you're standing up and using ghetto tools, you get dropped to 50% step success rate chance for the being standing, and then another 33% drop for using ghetto tools. Therefore it's gonna take some extra time unless you're carrying around a surgery belt or some shit.

**Comment by Xhuis** [comment]
thumbsdown

**Comment by RemieRichards** [comment]
thumbsdown

**Comment by Xhuis** [comment]
# don't steal my memes

**Comment by RemieRichards** [comment]
It's meme recycling, for a good cause, stopping goof, yet again.

**Comment by Iamgoofball** [comment]
i can't hold all these memes

**Comment by Bawhoppen** [comment]
can i meme too because i believe in the cause

**Comment by Zilenan91** [comment]
@Iamgoofball Actually using ghetto tools, some surgeries are impossible on the floor due to the chances being too low. I think embryo removal might be possible if on a table though, so... yeah blacklist it.

**Comment by SpookyDoc** [comment]
Yes doctors need to be more pointless :-1:

**Comment by Incoming5643** [comment]
I don't mind self surgery, but trying to remove the delay just kind of dragged this pull from "arguable" to "people are going to post giant thumbs down"

**Comment by Shadowlight213** [comment]
I can meme 2!
:+1: On self surgery
On delay removal

**Comment by TheDreamweaver** [comment]
AS someone who has attempted surgery on he station, the waits are absolute shit and an immediate turn off to surgery (which is why no one even does it anymore). For everyone saying it makes things to fast, compare it to how fast the rest of the game goes.

**Comment by Incoming5643** [comment]
Playing a round of spessmen isn't supposed to be a race, a lot of people just seem to have lost sight of this.

**Comment by PKPenguin321** [comment]
holy fuck i dont know how you guys can hate this this is actually fucking radical

**Comment by Incoming5643** [comment]
!!Surgery is not supposed to be a combat tactic!!

**Comment by bgobandit** [comment]
Combat surgery is powergaming on the level of tactical chairs.

:-1: x 1000

**Comment by bgobandit** [comment]
getting rid of delays is also stupid. this isn't Instant Gratification Spessmen you fucks

**Comment by PKPenguin321** [comment]
you guys are being silly
i'd like to see it at least tried in game

**Comment by TheDreamweaver** [comment]
ME TOO.

**Comment by Fox-McCloud** [comment]
What? The times for surgery are already ridiculously sort with TG surgeries---I don't really agree with shortening them at all.

Self-surgery is a BIG no-no, if only for the fact to prevent augmentation surgery.

**Comment by PKPenguin321** [comment]
fuck off fox we had self surgery for ages and everybody loved it but it was a """"""bug"""""" so it got removed

**Comment by Fox-McCloud** [comment]
And thanks goodness for that. The whole point of the implants being implants over something you just inject into yourself is to encourage co-operation and to curb science's power, if only a little. Yes, I'm sure every powergamer and their mother loved it.

It also means that doctors have a purpose other than just purely healing people 24/7.

**Comment by PKPenguin321** [comment]
nobody even uses augs they got nerfed to shit lol

and if by "every powergamer and their mother" you mean everybody that plays here except MrPerson, then yes, we did love it as a feature

and if self surgery really is as bad as you are putting it (it isn't), then it's a miracle our server isn't already dead from the massive amount of imbalance we had when it was in for several months

i'm pretty sure you don't even play here why do you care

**Comment by collinxchu** [comment]
Hahaha holy shit goofball are you capable of pushing anything that DOESN'T break balance? I mean I have my gripes with your chems but this is just laughably shit-tier, and it should be plainly obvious why.

**Comment by Chnkr** [comment]
I actually like this change, it has worked at goon for a long time, and i'd suggest you guys try it.

**Comment by Bawhoppen** [comment]
don't try and use the goon argument

**Comment by collinxchu** [comment]
@Chnkr Yeah I mean I like the idea of lessening delays and allowing self-surgery, but the former went a bit too far considering this means you can now theoretically pull someone's brain out if you stun them.

**Comment by Chnkr** [comment]
@collinxchu self sugery is a bad thing tough, maybe only allow it when drunk?

**Comment by CPTANT** [comment]
Yes on self surgery. 
No to delay removal.
Shortening them is ok, they are pretty long but don't fucking remove them.

**Comment by Razharas** [comment]
The self surgery was removed together with inability to hold things in hands while resting
Why cant we hold things in hands while willfully resting again?

**Comment by ninjacha** [comment]
/tg/ is suppose to be fun! YES TO EVERYTHING!

File: code/modules/mob/living/carbon/carbon_defense.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/modules/mob/living/carbon/carbon_defense.dm]: too bad it's the only way to do it without readding holding shit while resting

**Comment by Nanotransistor** [comment]
If you want to remove the delays you should add a lot more additional steps for each surgery to compensate.

**Comment by Qbopper** [comment]
why the fuck would you even think this is a good idea

it's fucking SURGERY, of course it's gonna take time

I have no strong opinions on self surgery, I would be perfectly happy with it being in or out, but removing the delay is just a terrible idea - remie's idea of "cut the delay in half" is a good compromise but by the sounds of it I don't think goof is gonna listen to fucking anyone, which is extremely disappointing

**Comment by Iamgoofball** [comment]
# the plan
## re-add delays at half speed

y/n

**Comment by Xhuis** [comment]
I think that the delays are fine as-is on the current servers. Surgery shouldn't be fast or instant.

**Comment by PKPenguin321** [comment]
im all for shortening delays, :+1:

**Comment by bgobandit** [comment]
delays are fine as they are

**Comment by Iamgoofball** [comment]
@Razharas the hell that would be required to get that working is not in my interests

**Comment by Supermichael777** [comment]
half speed is agreeable, pulling you brain out in front of the roboticist is just funny.:+1:

**Comment by Cookiesndip** [comment]
I'll reiterate something I said in the feedback thread.

+++ for self-surgery. Added so many different tricks, made augmentations and implants actually vaguely viable for antags despite taking longer to implement, permitted changing your identity if you had the right tools and a good deal of time, allowed you to hide things in yourself for permabrigging and so forth. There was *so* much that you could do that is simply impossible right now.

Removing it basically removed a whole chunk of antag potential, and removed the viaibility of implants for antags. Seriously, why would ANY antag trust themselves with an unknown factor that can kill them, accidentally or not during the process with zero resistance, on top of requiring a second willing person to do it in the first place. To make things worse, the implants require quite high tech levels along with minerals to get anything, which are sparsely used at best in the first place because it requires so many different factors to converge to obtain and implant, and even then it's an extraordinarily risky procedure that frankly, isn't worth the risk or time investment. Basic augments were nerfed to the ground anyway, requiring ridiculous amounts of fuel/wire to heal even the simplest damage.

Then cavity implants are literally never used these days, ESPECIALLY by antags because saying something like 'implant this bomb into me' is an easy way of getting yourself lynched by a nearby validhunter. That said, it's not like if self-surgery were back it wouldn't be fucking obvious to the room that you're implanting a bomb, but at least you can seek out a safe place to operate on yourself for a few minutes, at the cost of being crazy vulnerable over that period. Seems balanced to me.

If you want, remove the delay when operating on a person, but keep the delay when operating on yourself. Boom, balance. Time (duration of surgery), risk (vulnerability), and luck (minerals, R&D, etc.) for benefit seems perfectly fair.

**Comment by Core0verload** [comment]
I already reduced surgery delays in one of my updates. There is no point in removing them or reducing them further.

**Comment by GunHog** [comment]
Make "Self" surgery require a machine. Go build a hidden one in the antag surgery room in maint! Watching some guy walk into robotics and augment himself where he stands is just plain wacky and nonsensical.

**Comment by Lumipharon** [comment]
Stop removing shit that makes people have to actually rely on other players, jesus christ.
Why are we playin a multiplayer game if we just keep removing all the player interactions?

If antags can merrily get keys an work together 99% of the time (I have yet to personally see traitors EVER betray each other when using keys, except of course DA), then they can trust each other doing surgery.

**Comment by PKPenguin321** [comment]
clearly lumi has never got appendectus as a traitor while being on the run from sec and having no friends to help him
and having to sweatily cut myself open in a dark corner of medbay while xenos stalk the halls is one my  favorite memories of this game
there is nothing wrong with self surgery

**Comment by Lumipharon** [comment]
That's because appendicitis is fucking purely RNG bullshit - it's got nothing to do with self surgery.

**Comment by Iamgoofball** [comment]
appendix is dumb and needs removed but thats for another PR

**Comment by Saegrimr** [comment]
I hope this is face hugger stun times all over again, that was fun to watch.

**Comment by optimumtact** [comment]
goof you gotta split your PR's up into smaller parts, there are some good ideas in here that could be saved.

File: code/modules/mob/living/carbon/carbon_defense.dm (hunk 1 above)
Comment (optimumtact) [inline code/modules/mob/living/carbon/carbon_defense.dm]: we dont' do shit fixes and features goof, because we're not on a time budget

**Comment by Steelpoint** [comment]
When we had self surgery, which we had for years, it was rarely done.

I'm in favour of returning self surgey, if only because that's the exact kind of whacky thing I would expect a 2db spess man to do in ss13.

**Comment by Miauw62** [comment]
without surgery delay there's literally no point in not using ghetto tools, :-1:

**Comment by Iamgoofball** [comment]
@Miauw62 
surgery delay is coming back, read the big bold text

also, I don't quite understand why failure chance doesn't do some damage too since you fucked up the procedure

**Comment by Bawhoppen** [comment]
this is still dumb

**Comment by Strathcona** [comment]
Some sort of self-surgery machine would be ideal I think. All the fun possibilities with no cooperation required but it could be reliably gated with either your ability to break into the machine room in medbay or your ability to steal a board for it. The key thing is to prevent self surgery in most ridiculous situations without it being snowflakey or blacklisty.

**Comment by Supermichael777** [comment]
a surgery machine you had to program yourself might be cool.
or a parts implementation where higher tier parts unlock more/faster surgeries
. Putting removed things into sample jars like alien embryos for science/sale!

**Comment by TheDreamweaver** [comment]
God, you guys make it so that any coder with a bright idea is gonna run to the corner instead of the presentation table. I can guarantee you that if it was presented to the actual playerbase, you would get a postive majority response. If you guys stopped trying to shove your balancing sticks up everyone's ass for a moment and thought about what might be fun to try in a game of robust 2d spessmen, I'd think you get some great results. Not to say that sometimes this can lead to shit results,but then all you have to do is remove the specific thing leading to those results and everyone remains happy and creative.

**Comment by Bawhoppen** [comment]
@TheDreamweaver are you retarded?
if we had every feature a player ever wanted, we'd have 5x sanic infinite range eguns/tasers with infinite parasting and parapen c4 bullshit. If we had that kind of bullshit we'd be lifeweb-tier community (aka SHIT)




gonna regret posting this comment eventually

**Comment by TheDreamweaver** [comment]
Dude I said the majority of the playerbase. I can guarantee you their reaction to self surgery a me halved surgery times would be way different then a purposefully overpowered meme gun. One of those the playerbase would get fucking tired of real fast, and it ain't the surgery. There is a huge difference between something designed to be OP and something that could, in specific situations, be OP. Stop making shit comparisons and hear people out for once.

**Comment by Bawhoppen** [comment]
what
are you okay
did you take your meds this morning

**Comment by Core0verload** [comment]
bbut surgery times are already halved by my surgery update!

**Comment by TheDreamweaver** [comment]
@Bawhoppen Wow, you're a pleasure to converse with.

Whatever, if my opinion isn't shared by anyone, that we should prioritize encouraging creativity in coders, then I'll just stop contributing to things like this.

**Comment by TheDreamweaver** [comment]
@Core0verload Was not aware. Is your update already merged?

**Comment by Core0verload** [comment]
Merged a long time ago. I reduced delays for common actions (sawing, cutting, clamping, etc), but not for surgery-specific actions like face reshaping or eye fixing.

**Comment by Bawhoppen** [comment]
@TheDreamweaver I'm not even talking about disagreements, you literally made zero sense in your comment

**Comment by TheDreamweaver** [comment]
@Core0verload Hmm I wonder if goof is trying to half those too?

**Comment by Core0verload** [comment]
Goof is trying to remove all delays, according to code.

**Comment by TheDreamweaver** [comment]
@Bawhoppen Linguistically or logistically? And at what part? You're gonna have to help me out here.

**Comment by TheDreamweaver** [comment]
@Core0verload So I guess his new plan is to half everything. I guess due to your update, he would only need to half what you left out.

**Comment by optimumtact** [comment]
If you can't take criticism on your code, you should probably avoid developing.

Being critiqued on both code and idea are a necessary and valuable part of working on code.

Yes some people here can be very caustic and blunt when critiquing but you must look at it as a critique of code and not an attack on the person.

**Comment by Skullyton** [comment]
Instead of having it so you can just casually Kali Ma yourself while standing up, how about you can only do self surgery when you're buckled to a chair?

Also the delay's there because surgery takes time, you can't exactly rush that shit unless you've got high tech things that would negate the possibility of killing your patient by accidentally jolting a millimeter

**Comment by Firecage** [comment]
Powergamey PR is powergamey.

:-1:

**Comment by PKPenguin321** [comment]
firecage your post doesn't mean anything
if you make another post like that i'll powergamed the hell outoff you, you'd better watch out

**Comment by Bawhoppen** [comment]
@PKPenguin321 don't act like powergaming is good

**Comment by PKPenguin321** [comment]
the word powergaming by itself doesn't mean anything, it's what some people would call a "buzzword." now back off or i'll powergame you too :^)

>Do you seriously want to implant bombs into your own chests that badly, or change your own gender, or remove your own implants, or augment yourself? No, you shouldn't be able to.

why not? you present no good reason as to why none of these things should be done except that you don't want them to happen

**Comment by Lumipharon** [comment]
>Multiplayer game.
>Human interaction.

Naw, lets make medbay a self service centre, because fuck trusting doctors amirite?
Lets let people replace their own HEAD because fuck trusting roboticists/anyone with tools amirite?

**Comment by PKPenguin321** [comment]
>player interaction, muh realism
neither of these should take priority over a good gameplay feature, *especially* realism. player interaction i can see, but if your definition of interaction is sitting perfectly still while someone else silently clicks on you with various tools/you slowly die because you are a wanted traitor and got appendicitis, then your point really doesn't hold any weight

**Comment by optimumtact** [comment]
That makes appendicitis a bad gameplay feature, but surgery is designed as a lock on some more high end gear and fun and should not be made single user.

At least as far as I can tell, it's not a gameplay component itself because the simulation is at most skin deep (see what  I did there?)

I think goofs approach of allowing some basic self surgeries is going to best fit the situation here

**Comment by Iamgoofball** [comment]
appendicitis IS a bad gameplay feature oranges

**Comment by optimumtact** [comment]
I know

**Comment by Steelpoint** [comment]
Note to self, remove the appendicitis random event.

**Comment by Bawhoppen** [comment]
why do you nerds hate appendicitis
Not everything needs to be perfectly equal

**Comment by Xhuis** [comment]
Thinking back, the delay removal actually slightluy appeals to me. Tentative :+1: for that.

**Comment by Lumipharon** [comment]
The only time I have seen appendicitis and it hasn't been some awful piece of shit that no wanted, was when it happened to a lawyer during a hour long trial - and a doctor slammed him onto the table for an emergency procedure there and then.

**Comment by SweaterKittens** [comment]
Is it too late to hop aboard the "please god no" train?

https://cloud.githubusercontent.com/assets/5659269/10556677/198cf5b0-7458-11e5-8df6-b188820b7b7f.png

**Comment by PKPenguin321** [comment]
sweater, it didn't do that when we had it before
to you have anything that can actually substantiate those claims?

**Comment by Zilenan91** [comment]
this pr is getting to "IDs spawning in PDAs" levels

**Comment by Razharas** [comment]
Id consider it if it was made by letting people hold shit while resting
Like this it is just not good enough
Apply yourself

**Comment by Iamgoofball** [comment]
@Razharas good luck coding the snowflake for that

because so much in the game assumes that being on the floor or turned sideways = stunned

**Pull request closed without merging**